tests/migrations/test_commands.py

```diff
@@ -268,6 +268,42 @@ def test_showmigrations_plan(self):
         # Cleanup by unmigrating everything
         call_command("migrate", "migrations", "zero", verbosity=0)
 
+    @override_settings(MIGRATION_MODULES={"migrations": "migrations.test_migrations_plan"})
+    def test_migrate_plan(self):
+        """
+        Tests --plan output of migrate command
+        """
+        out = io.StringIO()
+        call_command("migrate", plan=True, stdout=out, no_color=True)
+        self.assertEqual(
+            "Planned operations:\n"
+            "migrations.0001_initial\n"
+            "    Create model Salamander --> \n"
+            "    Raw Python operation --> Grows salamander's tail\n"
+            "migrations.0002_second\n"
+            "    Create model Book --> \n"
+            "    Raw SQL operation --> SELECT * FROM migrations_book\n"
+            "migrations.0003_third\n"
+            "    Create model Author --> \n"
+            "    Raw SQL operation --> SELECT * FROM migrations_author\n",
+            out.getvalue()
+        )
+        call_command("migrate", "migrations", verbosity=0)
+
+        out = io.StringIO()
+        call_command("migrate", "migrations", "0001", plan=True, stdout=out, no_color=True)
+        self.assertEqual(
+            "Planned operations:\n"
+            "migrations.0003_third\n"
+            "    Undo Create model Author --> \n"
+            "    Raw SQL operation --> SELECT * FROM migrations_book\n"
+            "migrations.0002_second\n"
+            "    Undo Create model Book --> \n"
+            "    Raw SQL operation --> SELECT * FROM migrations_salamander\n",
+            out.getvalue()
+        )
+        call_command("migrate", "migrations", "zero", verbosity=0)
```

Comment:
The other test cases have a comment here:

```
# Cleanup by unmigrating everything
```

I think that's probably worthwhile (since I did a double-take myself reviewing even now).

Reply:
I'll add that , re-squash and get the commit message in line. Long time user, first time contributor (to django), so pardon the missteps, and thanks for your patience.